### PR TITLE
SPCTR-2759 - [Heading] Shift to a new paragraph block when pressing Enter key inside the heading block

### PR DESCRIPTION
**Pull request by supriyajjadhav**
### Description
[Heading] Shift to a new paragraph block when pressing Enter key inside the heading block

### Types of changes
New feature (non-breaking change which adds functionality) 

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Review by AbhijitNage123 — APPROVED** [review]

File: src/blocks/advanced-heading/render.js [hunk 1]
```diff
@@ -33,15 +35,15 @@ const Render = ( props ) => {
 	const headingText = (
 		<>
 			{ seperatorPosition === 'above-heading' ? separator : '' }
-			<Renderer { ...{ setAttributes, attributes, context } } />
+			<Renderer { ...{ setAttributes, attributes, context, onReplace, mergeBlocks } } />
```
Comment (rahulvarma722) [inline src/blocks/advanced-heading/render.js:38]: As we can see that this is the same object as line number 46 so can we define one variable at the top and use both places line number 38 and line number 46
Ex - const AnyName = { setAttributes, attributes, context, onReplace, mergeBlocks };

lline number 38, 46 
			<Component { ...AnyName } />

File: src/blocks/advanced-heading/renderer-desc.js [hunk 2]
```diff
@@ -26,6 +27,12 @@ const RendererDesc = ( props ) => {
 			className="uagb-desc-text"
 			onChange={ ( value ) => setAttributes( { headingDesc: value } ) }
 			allowedFormats={ allowedFormats }
+			onMerge={ mergeBlocks }
+			onSplit={( value ) => value ? 
+				createBlock('uagb/advanced-heading', {...props.attributes, headingDesc: value}) : createBlock('core/paragraph')}
```
Comment (rahulvarma722) [inline src/blocks/advanced-heading/renderer-desc.js:32]: Define the attributes at line number 7 and use only the attributes only here. and please maintain the pretty standard for now yourself spacially spacing.
EX - 				createBlock( 'uagb/advanced-heading',  { ...props.attributes, headingDesc: value } ) : createBlock( 'core/paragraph' ) }

File: src/blocks/advanced-heading/renderer-desc.js [hunk 3]
```diff
@@ -26,6 +27,12 @@ const RendererDesc = ( props ) => {
...
+			onReplace={ onReplace }
+			onRemove={ () => onReplace( [] ) }
+			{ ...props }
```
Comment (rahulvarma722) [inline src/blocks/advanced-heading/renderer-desc.js:35]: Please confirm once, Do we need to pass props in this component?

File: src/blocks/advanced-heading/renderer.js [hunk 4]
```diff
@@ -27,11 +30,16 @@ const Renderer = ( props ) => {
...
 			value={ headingTitle }
 			className="uagb-heading-text"
-			multiline={ false }
 			onChange={ ( value ) => {
 				setAttributes( { headingTitle: value } );
 			} }
 			allowedFormats={ allowedFormats }
+			onMerge={ mergeBlocks }
+			onSplit={( value ) => value ? 
+				createBlock('uagb/advanced-heading', {...props.attributes, headingTitle: value}) : createBlock('core/paragraph')}
```
Comment (rahulvarma722) [inline src/blocks/advanced-heading/renderer.js:39]: Please refer component RendererDesc comments

File: src/blocks/advanced-heading/render.js (hunk 1 above)
Comment (supriyajjadhav) [reply src/blocks/advanced-heading/render.js]: Done. Please check

File: src/blocks/advanced-heading/renderer-desc.js (hunk 2 above)
Comment (supriyajjadhav) [reply src/blocks/advanced-heading/renderer-desc.js]: Done. Please check

File: src/blocks/advanced-heading/renderer-desc.js (hunk 3 above)
Comment (supriyajjadhav) [reply src/blocks/advanced-heading/renderer-desc.js]: We can remove passing props to the component. Please check

File: src/blocks/advanced-heading/renderer.js (hunk 4 above)
Comment (supriyajjadhav) [reply src/blocks/advanced-heading/renderer.js]: Done

**Review by rahulvarma722 — APPROVED** [review]
Looks good to me

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**